bindings/python/sigrok/core/classes.i

```diff
@@ -548,4 +548,23 @@ std::map<std::string, Glib::VariantBase> dict_to_map_options(PyObject *dict,
 }
 }
 
+/* Return NumPy array from Analog::data(). */
+%extend sigrok::Logic
+{
+    PyObject * _data()
+    {
+        npy_intp dims[1];
+        dims[0] = $self->data_length();
+        int typenum = NPY_UINT8;
```

Comment:
It might be better to look at the unitsize of the logic data and return an array of the appropriate type/dimensions.
E.g. if using a 16-channel device with unitsize 2 returning 100 samples, that would give you a length 100 uint16 array rather than a length 200 uint8 array.

Reply:
i've also thought about this approach, but what happens with really big logic analyser? i don't know if uint128 or even unit256 is supported.

Reply:
Yeah, that's a good point.
I think the best solution then is to return a uint8 array of shape (N, M) where N is the number of samples and M is the logic data unitsize in bytes.

Reply:
done